test/unit/finance/bill_variable_for_plan_changed_test.rb

```diff
@@ -3,25 +3,33 @@
 require 'test_helper'
 
 class Finance::BillVariableForPlanChangedTest < ActiveSupport::TestCase
-  attr_reader :contract, :account, :app_plan
+  attr_reader :contract, :app_plan
 
   setup do
     @contract = FactoryBot.build_stubbed(:contract)
-    @account  = FactoryBot.build_stubbed(:simple_account)
+    account = FactoryBot.build_stubbed(:simple_account)
     @app_plan = FactoryBot.build_stubbed(:application_plan)
 
+    contract.stubs(:provider_account).returns(account)
+    account.stubs(:provider_can_use?).with(:instant_bill_plan_change).returns(true)
+
     @org_tz = ENV["TZ"]
   end
 
   teardown do
     ENV["TZ"] = @org_tz
   end
 
-  test "bill for variable" do
-    Timecop.travel(15.days.ago) if Time.now.mday == 1 # rubocop:disable Rails/TimeZone we don't use timezones in billing
+  test "bill for variable on 1st day of month UTC" do
+    ENV["TZ"] = "UTC"
+    Timecop.travel(Time.now.beginning_of_month) # rubocop:disable Rails/TimeZone we don't use timezones in billing
```

Comment:
We don't care about UTC, we care about local time. In the other test, timezone is changed to make sure calculations are done in local time instead of UTC. If local time is UTC, then we test nothing of that.

Current month in UTC should not matter at all. Perhaps if you use a timezone which is -6 for example, and set current month to current month UTC, then make sure that contract is billed as if it is previous month, it would be provide usefulness.